projects/angular/src/button/providers/button-group-focus-handler.service.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Injectable, Renderer2 } from '@angular/core';
+
+import { ClrPopoverToggleService } from '../../utils';
+import { FocusService } from '../../utils/focus/focus.service';
+import { FocusableItem } from '../../utils/focus/focusable-item/focusable-item';
+import { Linkers } from '../../utils/focus/focusable-item/linkers';
+import { InitialFocus } from './button-group-focus.enum';
+
+@Injectable()
+export class ButtonGroupFocusHandler {
+  private initialFocus: InitialFocus = InitialFocus.FIRST_ITEM;
+  private _unlistenFuncs: (() => void)[] = [];
+  private _menuOpened: boolean;
+  private buttons: FocusableItem[];
+  menuToggle: HTMLElement;
+  menu: HTMLElement;
+
+  get menuOpened(): boolean {
+    return this._menuOpened;
+  }
+
+  set menuOpened(value: boolean) {
+    this._menuOpened = value;
+    if (this._menuOpened) {
+      this.initialize();
+    }
+  }
+
+  constructor(
+    private focusService: FocusService,
+    private toggleService: ClrPopoverToggleService,
+    private renderer: Renderer2
+  ) {}
+
+  setInitialFocus(initialFocus: InitialFocus) {
+    this.initialFocus = initialFocus;
+  }
+
+  initialize() {
+    this.focusService.registerContainer(this.menu, '-1');
+    this.listenToKeys();
+    this.linkButtons();
+    switch (this.initialFocus) {
+      case InitialFocus.LAST_ITEM:
+        this.focusLastItem();
+        break;
+      default:
+        this.focusFirstItem();
+        break;
+    }
+  }
+
+  resetButtonsFocus() {
+    this.buttons.forEach(button => {
+      button.blur();
```

Comment:
Why is this needed? I'm just curious because I don't see what it does.

Reply:
We're setting "tabindex" to "-1" on blur, so we can have only one with 0 while navigating trough items and that's just resetting all buttons to 0. Using it two times, first when we are initialising the buttons trough link function because they don't have tabindex set to something by default and after closing the menu so they can go back to 0 again.